docs/modules/ROOT/pages/how-tos/configuring/adding-custom-tasks.adoc

```diff
@@ -0,0 +1,146 @@
+= Adding custom tasks
+
+If your pipeline uses xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts],
+you can add custom tasks to the pipeline without compromising the chain of trust.
+These tasks can do just about anything you'd like: set dynamic arguments for the build, run unit tests, send emails, etc.
+With one major restriction: they *cannot modify your source repo or the prefetched content* before the build.
+Doing so would violate the link:https://enterprisecontract.dev/docs/ec-policies/release_policy.html#trusted_task__trusted[Trusted Tasks] Enterprise Contract rule.
+
+What this means in Trusted Artifacts terms:
+
+* To operate on your repo, your Task must declare a `SOURCE_ARTIFACT` parameter and use a special step to download this artifact
+  (refer to the ready-to-use examples below).
+* Your Task must *not* return a modified `SOURCE_ARTIFACT`.
+  It's ok if the task modifies the source repo as it executes, but the modifications will be discarded.
+  The build will use the original, unmodified `SOURCE_ARTIFACT`.
```

Comment:
nit: 

* It is fine to generate the modified source artifact but that shouldn't be used in any future task that needs to be trusted.
* This is true for _any_ artifacts consumed (source or cachi2) by trusted tasks (prefetch or build)